presto-verifier/src/main/java/com/facebook/presto/verifier/Validator.java

```diff
@@ -227,7 +230,28 @@ private boolean validate()
 
         boolean matches = resultsMatch(controlResult, testResult, precision);
         if (!matches && checkDeterministic) {
-            return checkForDeterministicAndRerunTestQueriesIfNeeded();
+            matches = checkForDeterministicAndRerunTestQueriesIfNeeded();
+        }
+        if (matches && !dropTemporaryTables) {
+            QueryResult controlTearDownResult = tearDownQuery(
+                    queryPair.getControl(),
+                    controlGateway,
+                    controlUsername,
+                    controlPassword,
+                    controlTimeout,
+                    controlPostQueryResults,
+                    controlTeardownRetries);
+            QueryResult testTearDownResult = tearDownQuery(
+                    queryPair.getTest(),
+                    testGateway,
+                    testUsername,
+                    testPassword,
+                    testTimeout,
+                    testPostQueryResults,
+                    testTeardownRetries);
+            controlResult = controlTearDownResult.getState() == State.SUCCESS ? controlResult : controlTearDownResult;
```

Comment:
It might be better to just print a message if teardown fails rather than failing the result of validation.